autoPyTorch/data/tabular_target_validator.py

```diff
@@ -0,0 +1,265 @@
+import typing
+
+import numpy as np
+
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+import scipy.sparse
+
+import sklearn.utils
+from sklearn import preprocessing
+from sklearn.base import BaseEstimator
+from sklearn.exceptions import NotFittedError
+from sklearn.utils.multiclass import type_of_target
+
+from autoPyTorch.data.base_target_validator import BaseTargetValidator, SUPPORTED_TARGET_TYPES
+
+
+class TabularTargetValidator(BaseTargetValidator):
+    def _fit(
+        self,
+        y_train: SUPPORTED_TARGET_TYPES,
+        y_test: typing.Optional[SUPPORTED_TARGET_TYPES] = None,
+    ) -> BaseEstimator:
+        """
+        If dealing with classification, this utility encodes the targets.
+
+        It does so by also using the classes from the test data, to prevent encoding
+        errors
+
+        Arguments:
+            y_train (SUPPORTED_TARGET_TYPES)
+                The labels of the current task. They are going to be encoded in case
+                of classification
+            y_test (typing.Optional[SUPPORTED_TARGET_TYPES])
+                A holdout set of labels
+        """
+        if not self.is_classification or self.type_of_target == 'multilabel-indicator':
+            # Only fit an encoder for classification tasks
```

Comment:
Should we also raise an error if in regression the data type of the target is not integer?

Reply:
I think #85 can handle this. Now I do not know what you mean exactly, as we have different regression targets. Usually we expected regression to have float type, which is something that the code handles right now.

Reply:
Oh I meant if somehow someone gives y values as a string. Which does not contain numbers but words, by mistake. I think in this validator module we should take care of this as well